operator/apis/loki/v1beta1/rulerconfig_types.go

```diff
@@ -118,6 +118,37 @@ type AlertManagerSpec struct {
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Alert Relabel Configuration"
 	RelabelConfigs []RelabelConfig `json:"relabelConfigs,omitempty"`
+
+	// Client configuration for reaching the alertmanager endpoint.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="TLS Config"
+	Client *AlertManagerClientConfig `json:"client,omitempty"`
+}
+
+type AlertManagerClientConfig struct {
+	TLS        *AlertManagerClientTLSConfig  `json:"tls,omitempty"`
+	HeaderAuth *AlertManagerClientHeaderAuth `json:"headerAuth,omitempty"`
+	BasicAuth  *AlertManagerClientBasicAuth  `json:"basicAuth,omitempty"`
```

Comment:
These two are mutually exclusive right? If yes, adding validation at later stage would be good, e.g. towards making this API v1. For now adding comments will be ok.